### PR TITLE
feat: add new base slider component

### PR DESCRIPTION
**Pull request by victorcg88**
# Pull request template
 
Adds the stats to the search store and implements the new `BaseSlider` component.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [ ] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [ ] `Main`
- [X] Other. Specify: https://github.com/empathyco/x/tree/feat/add-stats-to-adapter

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

**Comment by annacv** [comment]
Hi, as a general observation, XDS is a system based on multiples of 6, so I will suggest some changes on odd values 🙏

File: packages/x-components/src/components/base-slider.vue [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      };
+    }
+  });
+</script>
+
+<style lang="css">
+  @import 'nouislider/dist/nouislider.css';
+
+  .x-base-slider {
+    gap: 15px;
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:109]:
```suggestion
    gap: 16px;
```

File: packages/x-components/src/components/base-slider.vue [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  .x-base-slider-selected-max {
+    display: flex;
+    justify-content: stretch;
+    align-items: stretch;
+    flex-flow: column nowrap;
+  }
+
+  .x-base-slider-selected {
+    display: inline-flex;
+    gap: 5px;
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:123]:
```suggestion
    gap: 4px;
```

File: packages/x-components/src/components/base-slider.vue [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+  .x-base-slider-selected-min,
+  .x-base-slider-selected-max {
+    flex: 50%;
+    gap: 8px;
+  }
+
+  .x-slider {
+    height: 4px;
+    padding: 0 15px;
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:134]:
```suggestion
    padding: 0 16px;
```

File: packages/x-components/src/components/base-slider.vue [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    box-shadow: none;
+    cursor: pointer;
+  }
+
+  .x-slider .noUi-handle {
+    height: 30px;
+    width: 30px;
+    top: -13px;
+    right: -15px;
+    padding: 5px;
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:147]:
```suggestion
    height: 24px;
    width: 24px;
    top: -10px;
    right: -10px;
    padding: 4px;
```

File: packages/x-components/src/components/base-slider.vue [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+  .x-slider .noUi-handle-lower {
+    padding-right: 7px;
+  }
+
+  .x-slider .noUi-touch-area {
+    transform: rotate(90deg);
+  }
+
+  .x-slider .noUi-handle-upper {
+    padding-left: 7px;
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:168]:
```suggestion
    padding-left: 4px;
```

File: packages/x-components/src/components/base-slider.vue [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    cursor: pointer;
+  }
+
+  .x-slider .noUi-handle:before,
+  .x-slider .noUi-handle:after {
+    content: none;
+  }
+
+  .x-slider .noUi-handle-lower {
+    padding-right: 7px;
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:160]:
```suggestion
    padding-right: 4px;
```

File: packages/x-components/src/components/base-slider.vue [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+  }
+
+  .x-base-slider-selected-min,
+  .x-base-slider-selected-max {
+    flex: 50%;
+    gap: 8px;
+  }
+
+  .x-slider {
+    height: 4px;
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:133]:
```suggestion
    height: 8px;
```

File: packages/x-components/src/components/base-slider.vue [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+      return {
+        slider,
+        rangeSelected
+      };
+    }
+  });
+</script>
+
+<style lang="css">
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:105]: maybe scoped?

File: packages/x-components/src/components/base-slider.vue [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+    display: inline-flex;
+    gap: 5px;
+  }
+
+  .x-base-slider-selected-min,
+  .x-base-slider-selected-max {
+    flex: 50%;
+    gap: 8px;
+  }
+
```
Comment (annacv) [inline packages/x-components/src/components/base-slider.vue:131]:
```suggestion

.x-base-slider-selected-min {
   align-items: start;
}

.x-base-slider-selected-max {
   align-items: end;
}
```

File: packages/x-components/src/components/base-slider.vue (hunk 3 above)
Comment (victorcg88) [reply packages/x-components/src/components/base-slider.vue]: I removed this customization, which, would need to be applied in the client app.

File: packages/x-components/src/components/base-slider.vue [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+  .x-slider {
+    height: 4px;
+    padding: 0 15px;
+    margin: 15px 0;
```
[the remark this answers is not in the document]
Comment (victorcg88) [reply packages/x-components/src/components/base-slider.vue]: As above, I removed this customization, which, would need to be applied in the client app.

File: packages/x-components/src/components/base-slider.vue (hunk 5 above)
Comment (victorcg88) [reply packages/x-components/src/components/base-slider.vue]: I removed this customization.

File: packages/x-components/src/components/base-slider.vue (hunk 6 above)
Comment (victorcg88) [reply packages/x-components/src/components/base-slider.vue]: I removed this customization.

File: packages/x-components/src/components/base-slider.vue (hunk 7 above)
Comment (victorcg88) [reply packages/x-components/src/components/base-slider.vue]: I removed this customization.

File: packages/x-components/src/components/base-slider.vue (hunk 8 above)
Comment (victorcg88) [reply packages/x-components/src/components/base-slider.vue:147]: No, it needs to be not scoped, so `nouislider` styles import works as needed.

Comment (victorcg88) [reply packages/x-components/src/components/base-slider.vue:147]: It needs to be not scoped, so the styles are correctly applied to the slider.

**Review by annacv — APPROVED** [review]
🎉

**Pull request merged**